spacy/ru/__init__.py

```diff
@@ -0,0 +1,64 @@
+# encoding: utf8
+from __future__ import unicode_literals, print_function
+
+from ..language import Language
+from ..attrs import LANG
+from ..tokens import Doc
+from .language_data import *
+
+
+class RussianTokenizer(object):
+    _morph = None
+
+    def __init__(self, spacy_tokenizer, cls, nlp=None):
+        try:
+            from pymorphy2 import MorphAnalyzer
+        except ImportError:
+            raise ImportError(
+                "The Russian tokenizer requires the pymorphy2 library: "
+                "try to fix it with "
+                "pip install pymorphy2==0.8")
+
+        RussianTokenizer._morph = RussianTokenizer._create_morph(MorphAnalyzer)
+
+        self.vocab = nlp.vocab if nlp else cls.create_vocab(nlp)
+        self._spacy_tokenizer = spacy_tokenizer
+
+    def __call__(self, text):
+        words = [self._normalize(RussianTokenizer._get_word(token))
+                 for token in self._spacy_tokenizer(text)]
+
+        return Doc(self.vocab, words, [False] * len(words))
```

Comment:
Thanks for all this! Definitely an exciting patch. I think the engine in `pymorphy2` might be useful for other languages as well.

Now that I'm rereading the patch, I think this line might have a bug. `[False] * len(words)` is setting the `spacy` attribute on the tokens, which sets whether the token 'owns' a trailing space. This is used to build the `doc.text` and `span.text` strings.

If you can't determine the spacing from the text, it's best to set this to `True`, not `False`. But to be fully correct, we should match into the string, and check whether there's a space. I haven't tested this, but the loop would look something like this:

```python

def get_spaces(token_strings, text):
    i = 0
    spaces = []
    for token in token_strings:
        i += len(token)
        if text[i] == ' ':
            spaces.append(True)
            i += 1
        else:
            spaces.append(False)
    return spaces
```

Reply:
Oh.. Thank you for review and suggestion! I'll think it over and fix, but not right now. In few hours. Again thank you.